### PR TITLE
#476: Adds bossy-form-selectmenu into bossy-form

### PR DESCRIPTION
**Pull request by tamfleet**

### CODE REVIEW

**Comment by lshort2** [comment]
I don't understand lines 53-61 of site/app/app.component.ts here?

My understanding was that we are refactoring the code to fit the new BossyFormInput, as discussed in Slack.

**Comment by tamfleet** [comment]
I am not quite sure what you are asking? The selectmenu was included in BossyFormInput for I added it into the interface in the input config.

File: src/config/form-input.ts [hunk 1]
```diff
@@ -36,7 +37,9 @@ export class BossyFormInputConfig {
 	public hasSuccess: boolean;
 	public hasWarning: boolean;
 	public hasDanger: boolean;
+	public selectmenu: BossyFormSelectMenuConfig
```
Comment (tabersays) [inline src/config/form-input.ts]: Add a semi colon

File: src/config/form-selectmenu.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export interface MenuItem {
+	value: string;
+	isDisabled?: boolean;
+}
+
+export class BossyFormSelectMenuConfig {
```
Comment (tabersays) [inline src/config/form-selectmenu.ts]: Refactor this config to match the same style as the form-input config. Have an interface for the config itself move all the parameters to the interface, have the class accept that interface object assign the values to this and make sure the are publicly assessable from the config class.

File: src/config/form-input.ts (hunk 1 above)
Comment (tabersays) [inline src/config/form-input.ts]: Add this to the interface above also.

File: site/app/app.component.ts [hunk 3]
```diff
@@ -14,6 +14,9 @@ import {BossyDropdown} from '../../dist/components/dropdown';
 import {BossyDropdownConfig} from '../../dist/config/dropdown';
 import {BossyDropdownMenuItem} from '../../dist/components/dropdown-menu';
 import {BossyDropdownMenuItemConfig} from '../../dist/config/dropdown-menu';
+import {BossyFormSelectMenu} from '../../dist/components/form-selectmenu';
+import {BossyFormSelectMenuConfig} from '../../dist/config/form-selectmenu';
+import {MenuItem} from '../../dist/config/form-selectmenu';
```
Comment (glawson) [inline site/app/app.component.ts]: Is this used anywhere in this class?

File: site/app/app.component.ts [hunk 4]
```diff
@@ -63,32 +67,52 @@ export class AppComponent {
...
-		const bossyRadioConfig = new BossyRadioConfig(['lions', 'tigers', 'bears'], false, 'uniqueId');
-
-		this.configService.setConfig('calendarConfig', calendarConfig);
-		this.configService.setConfig('formConfig', formConfig);
-		this.configService.setConfig('bossyRadioConfig', bossyRadioConfig);
-
 		const formInputConfig = new BossyFormInputConfig(formInput4);
-		const dropdownConfig = new BossyDropdownConfig('Dropdown Menu', 'button',
+		const bossyRadioConfig = new BossyRadioConfig(['lions', 'tigers', 'bears'], false, 'uniqueId');
+		const dropdownConfig = new BossyDropdownConfig( 'Dropdown Menu', 'button',
```
Comment (glawson) [inline site/app/app.component.ts]: BossyDropdownConfig( 'Dropdown Menu'  
                                        ^ extra space between ( and 'Dropdown

File: src/templates/form-input.html [hunk 5]
```diff
@@ -8,6 +8,9 @@
     <div *ngSwitchCase="'email'">
         <input type="email">
     </div>
+		<div *ngSwitchCase="'selectmenu'">
```
Comment (glawson) [inline src/templates/form-input.html]: extra tab

File: src/templates/form-selectmenu.html [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<select class="custom-select">
+  <option selected>{{config.title}}</option>
+	<option *ngFor="let item of config.items" value=item.value [disabled]=item.isDisabled >{{item.value}}</option>
```
Comment (glawson) [inline src/templates/form-selectmenu.html]: extra tab

**Comment by glawson** [comment]
closes #476

**Pull request merged**